src/Symfony/Component/Uid/Factory/TimeBasedUuidFactory.php

```diff
@@ -13,9 +13,15 @@
 
 use Symfony\Component\Uid\TimeBasedUidInterface;
 use Symfony\Component\Uid\Uuid;
+use Symfony\Component\Uid\UuidV1;
+use Symfony\Component\Uid\UuidV6;
+use Symfony\Component\Uid\UuidV7;
 
 class TimeBasedUuidFactory
 {
+    /**
+     * @var class-string<UuidV1|UuidV6|UuidV7>
```

Comment:
this should actually be added on the constructor parameter as well. Otherwise it won't help IDEs and static analysis tools to highlight bad usages.